subsys/testsuite/ztest/src/ztest_posix.c

```diff
@@ -0,0 +1,219 @@
+
+static const char *test_args;
+static bool list_tests;
+
+static void add_test_filter_option(void)
+{
+	static struct args_struct_t test_filter_s[] = {
+		/*
+		 * Fields:
+		 * manual, mandatory, switch,
+		 * option_name, var_name ,type,
+		 * destination, callback,
+		 * description
+		 */
+		{ false, false, true, "list", NULL, 'b', (void *)&list_tests, NULL,
+		  "List all suite and test cases" },
+		{ false, false, false, "test", "suite::test", 's', (void *)&test_args, NULL,
+		  "Name of tests to run. Comma separated list formatted as "
+		  "\'suiteA::test1,suiteA::test2,suiteB::*\'. An * can be used "
+		  "as a wildcard to run all tests within a suite." },
+		ARG_TABLE_ENDMARKER
+	};
+
+	native_add_command_line_opts(test_filter_s);
+}
+
+NATIVE_TASK(add_test_filter_option, PRE_BOOT_1, 10);
+
+/**
+ * @brief Try to shorten a filename by removing the current directory
+ *
+ * This helps to reduce the very long filenames in assertion failures. It
+ * removes the current directory from the filename and returns the rest.
+ * This makes assertions a lot more readable, and sometimes they fit on one
+ * line.
+ *
+ * Overrides implementation in ztest_new.c
+ *
+ * @param file Filename to check
+ * @returns Shortened filename, or @file if it could not be shortened
+ */
+const char *ztest_relative_filename(const char *file)
+{
+	const char *cwd;
+	char buf[200];
+
+	cwd = getcwd(buf, sizeof(buf));
+	if (cwd && strlen(file) > strlen(cwd) &&
+	    !strncmp(file, cwd, strlen(cwd)))
+		return file + strlen(cwd) + 1; /* move past the trailing '/' */
+	return file;
+}
+
+/**
+ * @brief Helper function to set list_tests
+ *
+ * @param value - Sets list_tests to value
+ */
+void ztest_set_list_test(bool value)
+{
+	list_tests = value;
+}
+
+/**
+ * @brief Helper function to get command line argument for listing tests
+ *
+ * @return true
+ * @return false
+ */
+bool z_ztest_get_list_test(void)
+{
+	return list_tests;
+}
+
+/**
+ * @brief Helper function to get command line test arguments
+ *
+ * @return const char*
+ */
+const char *ztest_get_test_args(void)
+{
+	return test_args;
+}
+
+/**
+ * @brief Lists registered unit tests in this binary one per line
+ *
+ * @return int Number of tests in binary
+ */
+int z_ztest_list_tests(void)
+{
+	struct ztest_suite_node *ptr;
+	struct ztest_unit_test *test = NULL;
+	int test_count = 0;
+
+	for (ptr = _ztest_suite_node_list_start; ptr < _ztest_suite_node_list_end; ++ptr) {
+		while ((test = z_ztest_get_next_test(ptr->name, test)) != NULL) {
+			TC_PRINT("%s::%s\n", test->test_suite_name, test->name);
+			test_count++;
+		}
+	}
+
+	/* List tests only once */
+	ztest_set_list_test(false);
+	return test_count;
+}
+
+/**
+ * @brief Returns true when s1 and s2 are identical in length and contents
+ *
+ * @param s1
+ * @param s2
+ * @return true
+ * @return false
+ */
+static bool z_ztest_strcmp_identical(const char *s1, const char *s2)
+{
+	return (strlen(s1) == strlen(s2) && strncmp(s1, s2, strlen(s1)) == 0);
```

Comment:
this is `!strcmp(s1, s2)` ... would recommend not making a helper for it as the not condition is typically obvious